File: AngularJS/src/org/angular2/entities/ivy/Angular2IvyDeclaration.java
// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package org.angular2.entities.ivy;

import org.angular2.entities.Angular2Declaration;
import org.jetbrains.annotations.NotNull;

public abstract class Angular2IvyDeclaration<T extends Angular2IvySymbolDef.Entity> extends Angular2IvyEntity<T>
  implements Angular2Declaration {
  public Angular2IvyDeclaration(@NotNull T entityDef) {
    super(entityDef);
  }

  @Override
  public boolean isStandalone() {
    return myEntityDef.isStandalone();
  }
}

File: Angular/src/org/angular2/entities/ivy/Angular2IvyDeclaration.kt
// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package org.angular2.entities.ivy

import org.angular2.entities.Angular2Declaration

abstract class Angular2IvyDeclaration<T : Angular2IvySymbolDef.Entity>(entityDef: T)
  : Angular2IvyEntity<T>(entityDef), Angular2Declaration {

  override val isStandalone: Boolean
    get() = myEntityDef.isStandalone
}
